bitnami/odoo/README.md

```diff
@@ -171,6 +243,80 @@ See the [Parameters](#parameters) section to configure the PVC or to disable per
+* Re-labeling so as to follow Helm label best practices (see [PR 3021](https://github.com/bitnami/charts/pull/3021))
+* Adaptation to use common Bitnami chart standards. The following common elements have been included: extra volumes, extra volume mounts, common annotations and labels, pod annotations and labels, pod and container security contexts, affinity settings, node selectors, tolerations, init and sidecar containers, support for existing secrets, custom commands and arguments, extra env variables and custom liveness/readiness probes.
+
+As a consequence, backwards compatibility from previous versions is not guaranteed during the upgrade. To upgrade to `9.0.0`, it should be done reusing the PVCs used to hold both the PostgreSQL and Odoo data on your previous release. To do so, follow the instructions below (the following example assumes that the release name is `odoo`):
+
+> NOTE: Please, create a backup of your database before running any of those actions.
+
+* Old version is up and running
+
+  ```console
+  $ helm ls
+  NAME	NAMESPACE	REVISION	UPDATED                              	STATUS  	CHART       	APP VERSION
+  odoo 	default  	1       	2020-10-21 13:11:29.028263 +0200 CEST	deployed	odoo-14.0.21	13.0.20201010
+
+  $ kubectl get pods
+  NAME                       READY   STATUS    RESTARTS   AGE
+  odoo-odoo-984f954b9-tk8t8   1/1     Running   0          16m
+  odoo-postgresql-0           1/1     Running   0          16m
+  ```
+
+* Export both database and Odoo credentials in order to provide them in the update
+
+  ```console
+  $ export POSTGRESQL_PASSWORD=$(kubectl get secret --namespace default odoo-postgresql -o jsonpath="{.data.postgresql-password}" | base64 --decode)
+
+  $ export ODOO_PASSWORD=$(kubectl get secret --namespace default odoo-odoo -o jsonpath="{.data.odoo-password}" | base64 --decode)
+  ```
+
+* The upgrade to the latest (`15.X.X`) version is going to fail
+
+  ```console
+  $ helm upgrade odoo bitnami/odoo --set odooPassword=$ODOO_PASSWORD --set postgresql.postgresqlPassword=$POSTGRESQL_PASSWORD
+  Error: UPGRADE FAILED: cannot patch "odoo-odoo" with kind Deployment: Deployment.apps "odoo-odoo" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/instance":"odoo", "app.kubernetes.io/name":"odoo"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
+  ```
+
+* Delete both the statefulset and recplicaset (PostgreSQL and Odoo respectively). Notice the option `--cascade=false` for the former.
+
+  ```console
+  $ kubectl delete deployment.apps/odoo-odoo
+  deployment.apps "odoo-odoo" deleted
+
+  $ kubectl delete statefulset.apps/odoo-postgresql --cascade=false
+  statefulset.apps "odoo-postgresql" deleted
+  ```
+
+* Now the upgrade works
+
+  ```console
+  $ helm upgrade odoo bitnami/odoo --set odooPassword=$ODOO_PASSWORD --set postgresql.postgresqlPassword=$POSTGRESQL_PASSWORD
```

Comment:
Didn't you need the existing PVC to upgrade postgres?

Reply:
Yes, but there is no need to specify it as in the previous step we have used `--cascade=false` with PostgreSQL. This way, the old pod (and the PVC) is preserved. Then, when the pod is deleted, a new one with the correct labels comes up and automatically re-uses the same PVC. Think of it as a failing pod that is re-scheduled by K8s, it will reuse the PVC 😄

This was derived from the [upgrading instructions](https://github.com/bitnami/charts/tree/master/bitnami/postgresql#900) in PostgreSQL Chart itself.